File: packages/eixam_connect_flutter/ios/Classes/EixamConnectFlutterPlugin.swift
import Flutter
import UIKit

public class EixamConnectFlutterPlugin: NSObject, FlutterPlugin {
  private var backgroundLocationBridge: BackgroundLocationBridge?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let plugin = EixamConnectFlutterPlugin()
    ProtectionRuntimeBridge.register(with: registrar)
    plugin.backgroundLocationBridge =
      BackgroundLocationBridge.register(with: registrar)
    FirmwareDfuBridge.register(with: registrar)
    SecureStorageBridge.register(with: registrar)
    registrar.publish(plugin)
  }

  public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
    backgroundLocationBridge?.detach()
    backgroundLocationBridge = nil
  }
}

File: packages/eixam_connect_flutter/android/src/main/kotlin/dev/eixam/connect/flutter/EixamConnectFlutterPlugin.kt
package dev.eixam.connect.flutter

import android.content.Context
import dev.eixam.connect.flutter.dfu.FirmwareDfuBridge
import dev.eixam.connect.flutter.protection.ProtectionRuntimeBridge
import dev.eixam.connect.flutter.telemetry.BackgroundTelemetryBridge
import dev.eixam.connect.flutter.storage.SecureStorageBridge
import io.flutter.embedding.engine.plugins.FlutterPlugin

class EixamConnectFlutterPlugin : FlutterPlugin {
    private var applicationContext: Context? = null
    private var protectionBridgeGeneration: Long? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        applicationContext = binding.applicationContext
        protectionBridgeGeneration = ProtectionRuntimeBridge.register(
            messenger = binding.binaryMessenger,
            context = binding.applicationContext,
        )
        BackgroundTelemetryBridge.register(
            messenger = binding.binaryMessenger,
            context = binding.applicationContext,
        )
        FirmwareDfuBridge.register(
            messenger = binding.binaryMessenger,
            context = binding.applicationContext,
        )
        SecureStorageBridge.register(
            messenger = binding.binaryMessenger,
            context = binding.applicationContext,
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        applicationContext?.let {
            protectionBridgeGeneration?.let(ProtectionRuntimeBridge::unregister)
            BackgroundTelemetryBridge.unregister()
            FirmwareDfuBridge.unregister()
            SecureStorageBridge.unregister()
        }
        protectionBridgeGeneration = null
        applicationContext = null
    }
}
